index.js

```diff
@@ -426,38 +438,39 @@ class BitbucketScm extends Scm {
      * @return {Promise}                       Resolves to a decorated author with url, name, username, avatar
      */
     async _decorateAuthor({ username }) {
-        const token = await this._getToken();
-        const options = {
-            url: `${USER_URL}/${encodeURIComponent(username)}`,
-            method: 'GET',
-            token
-        };
+        try {
+            const token = await this._getToken();
+            const options = {
+                url: `${USER_URL}/${encodeURIComponent(username)}`,
+                method: 'GET',
+                token
+            };
 
-        const response = await this.breaker.runCommand(options);
-        const { body } = response;
+            const response = await this.breaker.runCommand(options);
+            const { body } = response;
 
-        if (response.statusCode === 404 && !username.match(/^\{.*\}/)) {
-            // Bitbucket API has changed, cannot use strict username request anymore, for now we will
-            // have to return a simple generated decoration result to allow all builds to function.
-            // We will only allow this if the username is not a {uuid} pattern. Since if this is a {uuid}
-            // pattern, this likely is a valid 404.
             return {
-                url: '',
-                name: username,
-                username,
-                avatar: ''
+                url: body.links.html.href,
+                name: body.display_name,
```

Comment:
Do we need any checks for undefined here?

Reply:
updated to use hoek